src/include/misc.h

```diff
@@ -257,15 +257,22 @@
 #endif
 
 #ifdef HAVE_TIMESTAMPS
-#define	WT_TIMESTAMP(x) (x)
-typedef uint8_t wt_timestamp_t[WT_TIMESTAMP_SIZE];
+#define	WT_TIMESTAMP_PTR(x) (&x)
```

Comment:
@agorrod, I took a run at this one yesterday.

First, I agree with you and @sulabhM, I couldn't find any way to do this without switching to passing around pointers, everything I tried generated a total mess.

After some thought, I think the simplest approach is to not try and use a union, but instead to use separate typedefs and hide the magic of acquiring a pointer to the timestamp in the `WT_TIMESTAMP` macro.

I've pushed a new branch, PR #3510, for your consideration.